kubernetes/agent.ts

```diff
@@ -44,14 +48,22 @@ export class PulumiSelfHostedAgentComponent extends pulumi.ComponentResource {
             stringData: {
                 "PULUMI_AGENT_TOKEN": args.selfHostedAgentsAccessToken,
             }
-        }, {parent: this});
+        }, { parent: this });
 
         this.agentServiceAccount = new kubernetes.core.v1.ServiceAccount("deployment-agent", {
             metadata: {
                 namespace: args.namespace.metadata.name,
                 labels: this.labels,
             },
-        }, {parent: this});
+        }, { parent: this });
+
+        this.workerServiceAccount = new kubernetes.core.v1.ServiceAccount("deployment-agent", {
```

Comment:
I'm not sure the agent deployment should be responsible for creating the service account, I think it might be better to just pass through the service account name and let the user bring their own, which they can set up however their cloud requires

Reply:
That make sense to me. I will get it updated.